client/src/pages/Feed.js

```diff
@@ -587,6 +583,12 @@ const Feed = (props) => {
     }
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
+  useEffect(() => {
+    setQueryKeysValue(history, {
+      near_me: searchRange,
+    });
+  }, [searchRange]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
This can probably be added to `setQueryFromState`
Not sure if that will cause any problems and the way you did it is good as well.